crates/terminal_view/src/terminal_panel.rs

```diff
@@ -1136,6 +1137,39 @@ impl Render for TerminalPanel {
                                 .swap(&terminal_panel.active_pane.clone(), &to);
                             cx.notify();
                         }
+                    }),
+                )
+                .on_action(cx.listener(|terminal_panel, action: &MoveItemToPane, cx| {
+                    let panes = terminal_panel.center.panes();
+                    let Some(target_pane) = panes.get(action.0).map(|p| (*p).clone()) else {
+                        return;
+                    };
+                    let source_pane = terminal_panel.active_pane.clone();
+                    terminal_panel.center.move_active_item(
+                        &source_pane,
+                        &target_pane,
+                        true,
+                        true,
+                        cx,
+                    );
+                }))
+                .on_action(cx.listener(
+                    |terminal_panel, action: &MoveItemToPaneInDirection, cx| {
+                        let source_pane = terminal_panel.active_pane.clone();
+                        let Some(target_pane) = terminal_panel
+                            .center
+                            .find_pane_in_direction(&source_pane, action.0, cx)
+                            .cloned()
```

Comment:
```suggestion
                            .cloned()
```

NIT: The other part does not need this, so we can also do it in a similar way?

Reply:
Not sure what you mean by this. What other part? I'll check the code In a bit

Reply:
There are two similar code pieces, the other in the workspace is just fine without the `clone`, placing its code inside `if let Some` — we can rewrite this one to match that style.

Reply:
Seems it has nothing to do with `let Some` vs `if let Some`.

Those are two different `find_pane_in_direction` one is in `pane_group`:

https://github.com/zed-industries/zed/blob/25970650a79fbcc3d6adc451dd2b532b21ac2979/crates/workspace/src/pane_group.rs#L157

And the one in `workspace` that calls one from the `pane_group` and adds `.cloned()` here:

https://github.com/zed-industries/zed/blob/25970650a79fbcc3d6adc451dd2b532b21ac2979/crates/workspace/src/workspace.rs#L2954-L2962

Your code btw :) 
![image](https://github.com/user-attachments/assets/6e773abe-7db1-4461-bb1e-db946e045445)

I did update my code with an `if let` though to match the style and because now that the logic was moved to the `pane_group` there is less reason to get curmudgeoned about indent levels and it removes `else { return }`.